src/rotary_encoder.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "pico/multicore.h"
+#include "hardware/gpio.h"
+#include "rotary_encoder.h"
+
+static int8_t rotation = 0;
+static uint8_t pin_a;
+static uint8_t pin_b;
+
+bool rotary_encoder_callback(repeating_timer_t *rt)
+{
+  static int8_t status_table[] = {0,1,-1,0,-1,0,0,1,1,0,0,-1,0,-1,1,0};
+  static int8_t prev_status = 0;
+  int8_t current_status;
```

Comment:
I don't investigate the cause.
Original code doesn't work in my board.
But I changed like this, it works. 
```suggestion
  prev_status <<= 2;
  prev_status |= gpio_get(pin_a) << 0 | gpio_get(pin_b) << 1;
  rotation += status_table[prev_status & 0b1111];
```